### ISSUE TITLE
[0.10.0-beta1] Only one or couple CQs are being executed, but not all

### ISSUE THREAD

**Issue by zstyblik** [influxdata/influxdb#5415]
Hello,

I've created quite a lot of CQs. They're pretty much same in the nature. The only difference is the data they're being executed on. Later on, I've noticed that only one CQ is being executed (over and over again). I thought that CQ service got stuck, so I've restarted InfluxDB, and true more than just one CQ was being executed after restart. However, watching logs more closely now, I can see only subset of CQs are being executed and that I'm pretty much seeing the same names. Moreover, I'd say that one CQ is being executed more and more often in favour of the others and that the result will be the same.

```
cq_trig_1h      CREATE CONTINUOUS QUERY cq_trig_1h ON telegraf BEGIN SELECT min(cosine) AS cosine_min, [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1h."default".trig FROM telegraf."default".trig GROUP BY time(5m), * END
cq_trig_1s      CREATE CONTINUOUS QUERY cq_trig_1s ON telegraf BEGIN SELECT min(cosine) AS cosine_min, [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1s."default".trig FROM telegraf."default".trig GROUP BY time(10s), * END
cq_trig_1m      CREATE CONTINUOUS QUERY cq_trig_1m ON telegraf BEGIN SELECT min(cosine) AS cosine_min,  [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1m."default".trig FROM telegraf."default".trig GROUP BY time(1m), * END
cq_trig_1d      CREATE CONTINUOUS QUERY cq_trig_1d ON telegraf BEGIN SELECT min(cosine) AS cosine_min, [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1d."default".trig FROM telegraf."default".trig GROUP BY time(15m), * END
```

Actually, it already has reached to the point when only one CQ is being executed over and over again.

```
[continuous_querier] 2016/01/21 08:58:32 executing continuous query cq_cpu_1s (2016-01-21 08:42:10 +0000 UTC to 2016-01-21 08:42:20 +0000 UTC)
[continuous_querier] 2016/01/21 08:58:40 executing continuous query cq_cpu_1s (2016-01-21 08:42:00 +0000 UTC to 2016-01-21 08:42:10 +0000 UTC)
[continuous_querier] 2016/01/21 08:58:49 executing continuous query cq_cpu_1s (2016-01-21 08:41:50 +0000 UTC to 2016-01-21 08:42:00 +0000 UTC)
[continuous_querier] 2016/01/21 08:58:57 executing continuous query cq_cpu_1s (2016-01-21 08:41:40 +0000 UTC to 2016-01-21 08:41:50 +0000 UTC)
[continuous_querier] 2016/01/21 08:59:05 executing continuous query cq_cpu_1s (2016-01-21 08:41:30 +0000 UTC to 2016-01-21 08:41:40 +0000 UTC)
[continuous_querier] 2016/01/21 08:59:14 executing continuous query cq_cpu_1s (2016-01-21 08:41:20 +0000 UTC to 2016-01-21 08:41:30 +0000 UTC)
[continuous_querier] 2016/01/21 08:59:22 executing continuous query cq_cpu_1s (2016-01-21 08:41:10 +0000 UTC to 2016-01-21 08:41:20 +0000 UTC)
[continuous_querier] 2016/01/21 08:59:31 executing continuous query cq_cpu_1s (2016-01-21 08:41:00 +0000 UTC to 2016-01-21 08:41:10 +0000 UTC)
[continuous_querier] 2016/01/21 08:59:39 executing continuous query cq_cpu_1s (2016-01-21 08:40:50 +0000 UTC to 2016-01-21 08:41:00 +0000 UTC)
```

**Comment by mjdesa** [comment]
@zstyblik what happens if you only start 3 of the 4 CQ's, removing the only one that shows up in your logs?

```
cq_trig_1h      CREATE CONTINUOUS QUERY cq_trig_1h ON telegraf BEGIN SELECT min(cosine) AS cosine_min, [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1h."default".trig FROM telegraf."default".trig GROUP BY time(5m), * END
cq_trig_1m      CREATE CONTINUOUS QUERY cq_trig_1m ON telegraf BEGIN SELECT min(cosine) AS cosine_min,  [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1m."default".trig FROM telegraf."default".trig GROUP BY time(1m), * END
cq_trig_1d      CREATE CONTINUOUS QUERY cq_trig_1d ON telegraf BEGIN SELECT min(cosine) AS cosine_min, [...], percentile(sine, 99.000) AS sine_p99 INTO telegraf_1d."default".trig FROM telegraf."default".trig GROUP BY time(15m), * END
```

**Comment by zstyblik** [comment]
@mjdesa such setup works.

**Comment by zstyblik** [comment]
... or at least so it seems.

**Comment by mjdesa** [comment]
@zstyblik so it seems like its the frequently running CQ the clobbers everything else. Would you agree with that?

**Comment by zstyblik** [comment]
@mjdesa yes, that's correct. I don't know whether CQs are being executed in parallel or serial one-by-one. But I guess the latter is true. So, CQ is executed, scheduler does some calculations, gets a list of CQs to schedule, may be sorts it, and there it is - it's time to schedule that CQ again despite it just has been executed, because `GROUP BY time()` is so small. But it's just a wild guess of mine.
Also, it's possible there is complete different issue, because it seems to me that CQ service/scheduler actually "locks up" on such query in an infinite must-execute-this-query loop.

**Comment by zstyblik** [comment]
@mjdesa What comes to my mind is, may be the execution of other queries depends on how fast is CQ executed(which also means load, writes, SELECT etc.). Also, it's possible that CQ with `time(10s)` doesn't make sense or not technically possible. In that case, it should result in an error and such CQ shouldn't be scheduled.

**Comment by mjdesa** [comment]
@zstyblik can you verify that this doesn't happen in `0.9.6-`

**Comment by zstyblik** [comment]
@mjdesa sure, it shouldn't be a problem. However, I will be able to test it with only one Telegraf instance at most.

**Comment by zstyblik** [comment]
@mjdesa yes, it seems the issue isn't present in v0.9.6.1. However, I had only one Telegraf instance to put data in. Also, I've seen the following error message for one of the CQs :)

```
[continuous_querier] 2016/01/27 13:55:26 error executing query: CREATE CONTINUOUS QUERY  [...] END: err = write failed: field overflow
```

I had the instance up for almost an hour and it seems that all CQs are being executed.

**Comment by KiNgMaR** [comment]
Same issue in 887df3c73069a11bd4b0753dc6f4348ccd2e4594. If we look at the time stamps, it looks like it's trying to backfill (which I never requested) instead of computing fresh data:

    [continuous_querier] 2016/02/12 03:47:02 executing continuous query cq_requests_mean_duration_15s (2016-02-11 20:10:45 +0100 CET to 2016-02-11 20:11:00 +0100 CET)
    [continuous_querier] 2016/02/12 03:47:19 executing continuous query cq_requests_mean_duration_15s (2016-02-11 20:10:30 +0100 CET to 2016-02-11 20:10:45 +0100 CET)

**Comment by KiNgMaR** [comment]
This is still happening in 2894234b1ec225752e9ac5bcda1e5cd5a971f87f - now no CQs are being executed at all... even after recreating all of them...

    [xxx ~]# grep continu /var/log/influxdb/influxd.log
    [continuous_querier] 2016/02/24 08:34:48 continuous query service terminating
    [continuous_querier] 2016/02/24 08:37:07 Starting continuous query service
    [continuous_querier] 2016/02/24 08:38:48 continuous query service terminating
    [continuous_querier] 2016/02/24 08:41:08 Starting continuous query service

    [xxx ~]# influx
    Visit https://enterprise.influxdata.com to register for updates, InfluxDB server management, and monitoring.
    Connected to http://localhost:8086 version 0.11.0.n1456214433
    InfluxDB shell 0.11.0.n1456214433
    > use xxx_prod
    Using database xxx_prod
    > show continuous queries
    name: xxx_prod
    ----------------
    name                                    query
    cq_xxx_rows_per_minute_total          CREATE CONTINUOUS QUERY cq_xxx_rows_per_minute_total ON xxx_prod BEGIN SELECT sum(num_xxxs) INTO xxx_prod."default".cq_xxx_rows_per_minute_total FROM xxx_prod."default".xxx_requests GROUP BY time(1m) END
    cq_tariff_requests_per_minute_method    CREATE CONTINUOUS QUERY cq_tariff_requests_per_minute_method ON xxx_prod BEGIN SELECT count("duration") INTO xxx_prod."default".cq_tariff_requests_per_minute_method FROM xxx_prod."default".tariff_requests GROUP BY time(1m), method END
    cq_xxx_requests_per_minute_server     CREATE CONTINUOUS QUERY cq_xxx_requests_per_minute_server ON xxx_prod BEGIN SELECT count("duration") INTO xxx_prod."default".cq_xxx_requests_per_minute_server FROM xxx_prod."default".xxx_requests GROUP BY time(1m), "server" END
    cq_xxx_requests_mean_duration_15s     CREATE CONTINUOUS QUERY cq_xxx_requests_mean_duration_15s ON xxx_prod BEGIN SELECT mean("duration") INTO xxx_prod."default".cq_xxx_requests_mean_duration_15s FROM xxx_prod."default".xxx_requests WHERE error = '0' GROUP BY time(15s) END
    cq_xxx_requests_min_duration_15s      CREATE CONTINUOUS QUERY cq_xxx_requests_min_duration_15s ON xxx_prod BEGIN SELECT min("duration") INTO xxx_prod."default".cq_xxx_requests_min_duration_15s FROM xxx_prod."default".xxx_requests WHERE error = '0' GROUP BY time(15s) END
    cq_xxx_requests_max_duration_15s      CREATE CONTINUOUS QUERY cq_xxx_requests_max_duration_15s ON xxx_prod BEGIN SELECT max("duration") INTO xxx_prod."default".cq_xxx_requests_max_duration_15s FROM xxx_prod."default".xxx_requests WHERE error = '0' GROUP BY time(15s) END
    cq_xxx_requests_95pct_duration_15s    CREATE CONTINUOUS QUERY cq_xxx_requests_95pct_duration_15s ON xxx_prod BEGIN SELECT percentile("duration", 95.000) INTO xxx_prod."default".cq_xxx_requests_95pct_duration_15s FROM xxx_prod."default".xxx_requests WHERE error = '0' GROUP BY time(15s) END

    > use _internal;
    Using database _internal
    > show measurements;
    name: measurements
    ------------------
    name
    cq
    httpd
    queryExecutor
    runtime
    shard
    subscriber
    tsm1_cache
    tsm1_filestore
    tsm1_wal
    udp
    write

    > select * from cq where time > now() - 1d;
    > select * from cq where time > now() - 2d;
    > select * from cq where time > now() - 3d;
    > select * from cq where time > now() - 4d;
    > select * from cq where time > now() - 5d;
    name: cq
    --------
    time                    clusterID               hostname        nodeID  queryOk
    1455868217000000000     605311852907240708      xxxxx           0       114060

What can I do to help in tracking this down?

**Comment by mark-rushakoff** [comment]
Please open a new issue if you're experiencing CQ issues in 1.1 or 1.2rc1.
